nemo/collections/llm/gpt/data/mock.py

```diff
@@ -9,6 +9,12 @@
 
 from nemo.lightning.pytorch.plugins import MegatronDataSampler
 
+HAVE_TE = True
+try:
+    import transformer_engine
+except (ImportError, ModuleNotFoundError):
```

Comment:
If the TE import fails, then the import error gets hidden in this except block. 

@marcromeyn @ericharper could you please share the right way to "import guard" in NeMo 2.0

Reply:
I agreed with @ashors1 that we can first merge this with the messy imports and later refactor them when we have better import guards.